### PR TITLE
feat: rodex use current mailbox opentype

### LINKED ISSUE #3715: [Feature Request] Rodex use openType
## ------------------ Openkore Issues Template ------------------

* **Openkore version git**: latest

* **Server**: all

* **Bug Report / Feature Request**:

when a rodex mail is sent by system , the type of mail is 1
but in Commands.pm
read mail packet is 
`$messageSender->rodex_read_mail(0,$arg2,0);`
it need to be
`$messageSender->rodex_read_mail(1,$arg2,0);`

it is same for delete and getitems

this is tested , wait for a fix 
And it seems the type of mail  is not processed in Receive.pm . Can bot get type of mail and use it in $messageSender? 
* **Summary**:

### PR DESCRIPTION
**Pull request by alisonrag**
this fixes #3715

### CODE REVIEW

**Comment by ya4ept** [comment]
Is this a new version of package? On the euRO server type is always 0.

**Comment by alisonrag** [comment]
> Is this a new version of package? On the euRO server type is always 0.

image

in image above the number is related to the tab, some servers like cRO and tRO has some stuffs like claim item from website, then the item go to other tabs

like the following:
image

image

**Comment by ya4ept** [comment]
There is only one RodEx button in the client, when it is pressed, type 0 is always opened.
How can we make `rodex open 2` via client?

**Comment by alisonrag** [comment]
got your point, seems like in newer version only 0 is sent

**Comment by alisonrag** [comment]
in newer versions type 0 is sent and for each mail received it has the opentype and client auto distribute it in the correct tabs:
https://github.com/OpenKore/openkore/blob/380fcd04ffd05cd0942da1ce809d88cad96d922e/src/Network/Receive.pm#L8382

for older versions still necessary send the opentype:
from rathena:
```/// 09e8 <mail tab>.B <mail id>.Q (CZ_OPEN_MAILBOX)
/// 0ac0 <mail id>.Q <unknown>.16B (CZ_OPEN_MAILBOX2)```

**Comment by alisonrag** [comment]
tested by user:
https://github.com/OpenKore/openkore/issues/3715#issuecomment-1421721481

**Comment by ya4ept** [comment]
`rodex maillist` before #[f558816](https://github.com/OpenKore/openkore/pull/3716/commits/f558816601b9888cb42139477675754fad34eb9a)
```
------------------------------ Rodex Mail Page 0 ------------------------------
0    From:  crima                     Read:   No   ID:  16971197  Title:  TITLE
1    From:  Admin                     Read:   Yes  ID:  2686200   Title:  Your items from 4Game.
2    From:  Admin                     Read:   No   ID:  2630053   Title:  Your items from 4Game.
-------------------------------------------------------------------------------
```

`rodex maillist` after:
```
------------------------------ Rodex Mail Page 0 ------------------------------
0   From: crima     Read: No  ID: 1697119 Title: TITLE
1   From: Admin     Read: Yes ID: 2686200 Title: Your items from 4Game.
2   From: Admin     Read: No  ID: 2630053 Title: Your items from 4Game.
-------------------------------------------------------------------------------
```

File: src/Commands.pm [hunk 1]
```diff
@@ -7485,9 +7493,9 @@ sub cmdRodex {
 			error T("You are not writing a rodex mail.\n");
 			return;
 
-		} elsif (!exists $rodexWrite->{zeny} || !exists $rodexWrite->{body} || !exists $rodexWrite->{title} || !exists $rodexWrite->{target}) {
+		} elsif (!exists $rodexWrite->{target}) {
 			error T("Error in function 'rodex send' (Send finished rodex mail)\n" .
-				"You still have to set something to send the mail (title, body, zeny or target)\n");
+					"You mast set target of rodex mail. Usage: rodex settarget <player_name|self>\n");
```
Comment (alisonrag) [inline src/Commands.pm:7498]: should be "must"

Comment (ya4ept) [reply src/Commands.pm]: done

**Comment by alisonrag** [comment]
seems fine to me

**Comment by ya4ept** [comment]
BEFORE:
```
rodex open
-------------------------------------------------- Rodex Mail Page  ---------------------------------------------------
0    From:  crima                     Read:   No   ID:  15679193  Title:  TITLE                     Content: Text
1    From:  Admin                     Read:   Yes  ID:  4567800   Title:  Your items from 4Game.    Content: Text, Item
-----------------------------------------------------------------------------------------------------------------------

rodex list
------------------------------- Rodex Mail List -------------------------------
  0 15679193  not read   From: crima                   Title: TITLE
  1 2565184   not read   From: Admin                   Title: Your items from 4Game.
-------------------------------------------------------------------------------

rodex maillist
------------------------------ Rodex Mail Page 0 ------------------------------
0   From: Admin     Read: Yes ID: 2565184 Title: Your items from 4Game. 
1   From: Admin     Read: No  ID: 2565185 Title: Your items from 4Game. 
-------------------------------------------------------------------------------
```

AFTER:
```
rodex open
-------------------------------------------------- Rodex Mail Page 0 --------------------------------------------------
 #  ID       From                    Att  New  Expire    Title
 0  15679193 crima                   -    Yes   15 Days  TITLE
 1  12346757 crima                   -    No    15 Days  sdfsdfsdfsdfsdf ;l;lk; sdfsdfsdf
 2  2565184  Admin                   i    No   363 Days  Your items from 4Game.
-----------------------------------------------------------------------------------------------------------------------
rodex list
--------------------------------------------------- Rodex Mail List ---------------------------------------------------
 #  ID       From                    Att  New  Expire    Title
 0  2565184  Admin                   i    Yes  264 Days  Your items from 4Game.
 1  15679193 crima                   -    No    14 Days  TITLE
 2  12346757 crima                   -    No    15 Days  sdfsdfsdfsdfsdf ;l;lk; sdfsdfsdf
-----------------------------------------------------------------------------------------------------------------------
rodex maillist
-------------------------------------------------- Rodex Mail Page 0 --------------------------------------------------
 #  ID       From                    Att  New  Expire    Title
 0  15679193 crima                   -    Yes   15 Days  TITLE
 1  12346757 crima                   -    No    15 Days  sdfsdfsdfsdfsdf ;l;lk; sdfsdfsdf
 2  2565184  Admin                   i    No   363 Days  Your items from 4Game.
-----------------------------------------------------------------------------------------------------------------------
```

**Comment by ya4ept** [comment]
@alisonrag  you can test the following scenario?
1. `rodex open`
2. `rodex read xxx`, where "xxx" is the email ID from the administrator
3. `rodex read yyy`, where "yyy" is the email ID from a normal user

I think you're defining the `$rodexCurrentType` variable incorrectly. Each mail has an individual type.

**Comment by alisonrag** [comment]
> @alisonrag you can test the following scenario?
> 
> 1. `rodex open`
> 2. `rodex read xxx`, where "xxx" is the email ID from the administrator
> 3. `rodex read yyy`, where "yyy" is the email ID from a normal user
> 
> I think you're defining the `$rodexCurrentType` variable incorrectly. Each mail has an individual type.

seems like it not work for new packets, but still works for old one, so is necessary to add new check on read and maybe delete

**Comment by ya4ept** [comment]
I will try to pour in my old code that works on euRO servers

**Comment by ya4ept** [comment]
@alisonrag  you can test the following scenario again?
1. `rodex open`
2. `rodex read xxx`, where "xxx" is the email ID from the administrator
3. `rodex read yyy`, where "yyy" is the email ID from a normal user

**Comment by ya4ept** [comment]
```
rodex list
--------------------------------------------------- Rodex Mail List ---------------------------------------------------
 #  ID       From                    Att  New  Expire    Title
 0  17007057 crima                   -    No     8 Days  sdfsdfsdfsdfsdf ;l;lk; sdfsdfsdf
 1  16971197 crima                   -    No     6 Days  TITLE
 2  2710242  Admin                   i    Yes  364 Days  Your items from 4Game.
 3  2699084  Admin                   i    No   363 Days  Your items from 4Game.
 4  2698181  Admin                   i    No   363 Days  Your items from 4Game.
 5  2697004  Admin                   -    No   361 Days  Your items from 4Game.
 6  2686200  Admin                   i    No   355 Days  Your items from 4Game.
 7  2630053  Admin                   i    Yes  323 Days  Your items from 4Game.
 8  2565184  Admin                   i    No   288 Days  Your items from 4Game.
 9  2556105  Admin                   i    Yes  281 Days  Your items from 4Game.
10  2511096  Admin                   -    No   256 Days  Your items from 4Game.
-----------------------------------------------------------------------------------------------------------------------

rodex read 10
-------------------------------- Mail 2511096 ---------------------------------
Mail type:   Account mail
From:        Admin
Title:       Your items from 4Game.
Message:     Get your items and continue adventuring!
Item count:  0
Zeny:        0
-------------------------------------------------------------------------------
```

**Comment by ya4ept** [comment]
BEFORE [41f5db1](https://github.com/OpenKore/openkore/pull/3716/commits/41f5db16cfdba212cadd85d25fda5ec978792fe5)
```
------------------------------------------------ Mail (2511096) Admin -------------------------------------------------
From: Admin
Message:
Get your items and continue adventuring!
Item count:   0
Zeny:         0
-----------------------------------------------------------------------------------------------------------------------
```

AFTER:
```
----------------------------------------------- Mail 2511096 from Admin -----------------------------------------------
Mail type:   Account mail
Title:       Your items from 4Game.
Message:     Get your items and continue adventuring!
Item count:  0
Zeny:        0
-----------------------------------------------------------------------------------------------------------------------
```

**Comment by ya4ept** [comment]
BEFORE #[312dc1e](https://github.com/OpenKore/openkore/pull/3716/commits/312dc1e1ec0954ab7a68c778c2f0a6a2dd24484e)
```
-------------------------------------------------- Rodex Mail Target --------------------------------------------------
Name:  Voices                    Base Level:  72   Class:  Soul Linker       Char ID:  552334
-----------------------------------------------------------------------------------------------------------------------
```

AFTER:
```
-------------------- Rodex Mail Target --------------------
Name: Voices                   Base Level: 72
Char ID: 552334                Class: Soul Linker
-----------------------------------------------------------
```

**Comment by ya4ept** [comment]
added a new command: **rodex draft**
```
------------------------------------------------ Draft mail for crima -------------------------------------------------
Recepient: crima                    Base Level: 1
  Char ID: 9313726                  Class: Novice
------
Title:      this is an example of a mail
Message:    Openkore is an advanced automated assistant (see Bot ) for the MMORPG Ragnarok Online . The program is
free, Open Source and mostly written in Perl , only some parts are written in C++ . The perl interpreter allows
cross-platform use ( Windows and Linux are officially supported).
-----------------------------------------------------------------------------------------------------------------------
-------------- Rodex mail item list --------------
-- Usable --

-- Equipment --
0    Knife [3] (Weapon)

-- Non-Usable --
--------------------------------------------------
```

**Comment by ya4ept** [comment]
@alisonrag 
i made all my fixes, please test it on other servers

**Review by LastRoze — APPROVED** [review]

**Comment by ya4ept** [comment]
Tested by @hubertyang88: https://github.com/OpenKore/openkore/issues/3715#issuecomment-1455747326

**Review by ya4ept — APPROVED** [review]

**Pull request merged**